### PR TITLE
[1LP][RFR] New proxy tests

### PR DESCRIPTION
**Pull request by otsuman**
{{pytest: -v cfme/tests/configure/test_proxy.py --use-provider ec2west }}

### CODE REVIEW

**Comment by lkhomenk** [comment]
with the [BZ](https://bugzilla.redhat.com/show_bug.cgi?id=1444048) - we might need one more verification in UI that advanced settings were added/edited/removed

**Comment by dajoRH** [comment]
Lint report for commit f4027188dfbb2aa55a4b98991758b5b9308a4d25:

`cfme/tests/configure/test_proxy.py`:
- :red_circle: Line 5:1: **F401** *'cfme.cloud.provider.CloudProvider' imported but unused*
- :red_circle: Line 10:1: **F401** *'cfme.utils.wait.RefreshTimer' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/tests/configure/test_proxy.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+    def _is_ip_in_log():
+        log_file = proxy_ssh.run_command("cat /var/log/squid/access.log").output
+        return appliance_ip in log_file
+    # need to wait until requests will occur in access.log or check if its empty after some time
+    wait_for(func=_is_ip_in_log, fail_condition=not valid_proxy, num_sec=180, delay=10)
+    assert _is_ip_in_log() == valid_proxy
+
+
+@pytest.yield_fixture(scope="function")
```
Comment (lkhomenk) [inline cfme/tests/configure/test_proxy.py]: REQUIRED: not needed anymore - kinda deprecated and replaced by  just `pytest.fixture`

**Comment by lkhomenk** [comment]
looks great. it was needed really! Thanks @otsuman ! waiting for PRT results

File: cfme/utils/appliance/__init__.py [hunk 2]
```diff
@@ -2050,6 +2050,21 @@ def update_advanced_settings(self, settings_dict):
...
+                    'port': port,
+                    'user': user,
+                    'password': password}
+        try:
+            vmdb_config['http_proxy'][proxy_type] = settings
+        except KeyError as ex:
+            logger.error('Incorrect provider type')
+            logger.exception(ex)
+            raise Exception('Impossible to create proxy with current provider type')
+        return self.update_advanced_settings(vmdb_config)
```
Comment (quarckster) [inline cfme/utils/appliance/__init__.py]: **QUESTION**
What should be returned here? `update_advanced_settings` returns nothing in https://github.com/ManageIQ/integration_tests/blob/master/cfme/utils/appliance/__init__.py#L2012

File: cfme/tests/configure/test_proxy.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    proxy_port = data['port']
+    prov = get_mgmt(proxy_provider_key)
+    deploy_template(proxy_provider_key,
+                    depot_machine_name,
+                    template_name=proxy_template_name)
+    yield prov.get_ip_address(depot_machine_name), proxy_port
+    prov.delete_vm(depot_machine_name)
+
+
+def validate_provider(provider, valid_proxy=True):
```
[the remark this answers is not in the document]
Comment (otsuman) [reply cfme/tests/configure/test_proxy.py]: This validate method just checking is provider refreshed or not and if not trying to refresh it. But I need a little bit different verification

**Comment by jkrocil** [comment]
`WebDriver` error, which is not happening locally, is blocking this. @quarckster Do you mind taking a look at this one?

**Comment by quarckster** [comment]
@jkrocil I'l take a look

**Pull request merged**